Docs: Improve guides for contributing (#19575)
grafana/grafana: CONTRIBUTING.md

# Contributing to Grafana

> Thanks for your interest in contributing to Grafana! We welcome all people who wants to contribute in a healthy and constructive manner within our community. To help us create a safe and positive community experience for all, we require all participants to adhere to the [Code of Conduct](CODE_OF_CONDUCT.md).

Reviewer: ```suggestion
Thank you for your interest in contributing to Grafana! We welcome all people who want to contribute in a healthy and constructive manner within our community. To help us create a safe and positive community experience for all, we require all participants to adhere to the [Code of Conduct](CODE_OF_CONDUCT.md).
```

This document is a guide to help you through the process of contributing to Grafana.

## Become a contributor

You can contribute to Grafana in several ways. Here are some examples:

- Contribute to the Grafana codebase.
- Report and triage bugs.
- Develop community plugins and dashboards.
- Write technical documentation and blog posts, for users and contributors.
- Organize meetups and user groups in your local area.
- Help others by answering questions about Grafana.

> For more ways you can contribute, check out the [Open Source Guides](https://opensource.guide/how-to-contribute/).

Reviewer: ```suggestion
For more ways to contribute, check out the [Open Source Guides](https://opensource.guide/how-to-contribute/).
```

Reviewer: More often than not, we can infer "you" in an imperative mood.

### Report bugs

Report a bug by submitting a [bug report](https://github.com/grafana/grafana/issues/new?labels=type%3A+bug&template=1-bug_report.md). Make sure that you provide as much information as possible on how to reproduce the bug.

> Before submitting a new issue, try to make sure someone hasn't already reported the problem. Look through the [existing issues](https://github.com/grafana/grafana/issues) for similar issues.

Reviewer: ```suggestion
Before submitting a new issue, ensure is not reported already. Look through the [existing issues](https://github.com/grafana/grafana/issues) for similarities.
```

Correct me, if I'm wrong but we want to don't want it to be optional.

Author: It's not, but sometimes it can be difficult to guarantee it hasn't already been reported. I'd prefer to err on the side of accepting here, to avoid missing out on issues. Is there a way a way to rephrase it, that's not as strict?

Reviewer: that's a good shout. Sadly, I don't have a good answer for that. Feel free to leave it as is.

#### Security issues

If you believe you've found a security vulnerability, please read our [security policy](https://github.com/grafana/grafana/security/policy) for more details.

### Suggest enhancements

If you have an idea of how to improve Grafana, submit an [enhancement request](https://github.com/grafana/grafana/issues/new?labels=type%3A+feature+request&template=2-feature_request.md).

### Your first contribution

Unsure where to begin contributing to Grafana? Start by browsing issues labeled `beginner friendly` or `help wanted`.

> - [Beginner-friendly](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) issues shouldn't require more than changing a small number of lines.

Reviewer: ```suggestion
- [Beginner-friendly](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) issues are generally are straightforward to complete.
```

What do you think about this phrasing? I don't think I keep in mind the number of lines to be changed as we label this issues but instead the complexity of it.

Author: I tried to be a bit more specific to avoid words like "simple" and "easy". I can't decide whether what's seen as "straightforward" by one, might not be for another.

Reviewer: I recognise this is a delicate topic. That's why I added the word _generally_ - I think this covers for those cases where one might disagree.

Reviewer: In my eyes, it reads "we think these are relatively straightforward but there's room for disagreement"

Author: ```suggestion
- [Beginner-friendly](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) issues are generally straightforward to complete.
```

Author: I'm fine with that, just removed an extra "are"

> - [Help wanted](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) are issues that are a bit more involved than `beginner friendly` issues.

Reviewer: ```suggestion
- [Help wanted](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issues are problems we would like the community to help us with regardless of complexity.
```

Would love @marefr's opinion here as this is my current understanding of this.

If you're looking to make a code change, see how to set up your environment for [local development](contribute/development.md).

> When you're ready to contribute your change, it's time to [Create a pull request](/contribute/pull-request.md).

Reviewer: ```suggestion
When you're ready to contribute, it's time to [Create a pull request](/contribute/pull-request.md).
```

#### Contributor License Agreement (CLA)

> Before we can accept your pull request, we need you to [submit a CLA](https://grafana.com/docs/contribute/cla/).

Reviewer: ```suggestion
Before we can accept your pull request, you need to [sign our CLA](https://grafana.com/docs/contribute/cla/). If you haven't, our CLA assistant will add a comment to your pull request prompting you to.
```

Author: ```suggestion
Before we can accept your pull request, you need to [sign our CLA](https://grafana.com/docs/contribute/cla/). If you haven't, our CLA assistant prompts you to when you create your pull request.
```

Reviewer: Looks good!

## Community

- Follow [@grafana on Twitter](https://twitter.com/grafana/)
- Read and subscribe to the [Grafana blog](https://grafana.com/blog/)
- If you have a specific question, check out our [discussion forums](https://community.grafana.com).
- For general discussions, join us on the [official Slack](http://slack.raintank.io/).

## Where do I go from here?

> - Set up your [development environment](contribute/development.md).

Reviewer: ```suggestion
- Set up your [development environment](contribute/development.md)
```

Author: I think it's actually the other items should end with a period.

Reviewer: Giving this a second read. I agree. Let me fix that!

> - Learn how to [contribute documentation](contribute/documentation.md)

Reviewer: ```suggestion
- Learn how to [contribute documentation](contribute/documentation.md).
```

> - Get started [developing plugins](https://grafana.com/docs/plugins/developing/development/) for Grafana

Reviewer: ```suggestion
- Get started [developing plugins](https://grafana.com/docs/plugins/developing/development/) for Grafana.
```
